### PR TITLE
Add Search folders functionality

### PR DESCRIPTION
**Pull request by adimiz1**
### Brief Summary of Changes
Add Search folders functionality.

#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [ ] Refactoring
- [x] New feature
- [ ] Bug fix
- [x] Adds more tests

#### Are tests included?
- [x] Yes
- [ ] No

#### Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

File: cloudinary-test-common/src/main/java/com/cloudinary/test/AbstractSearchTest.java [hunk 1]
```diff
@@ -60,6 +60,14 @@ public void shouldFindResourcesByTag() throws Exception {
         assertEquals(3, resources.size());
     }
 
+    @Test
+    public void shouldFindFolders() throws Exception {
+        cloudinary.api().createFolder("testFolder", null);
```
Comment (const-cloudinary) [inline cloudinary-test-common/src/main/java/com/cloudinary/test/AbstractSearchTest.java:65]: I would make the folder name unique, and delete it on tearDown.

**Review by const-cloudinary — CHANGES_REQUESTED** [review]

File: cloudinary-test-common/src/main/java/com/cloudinary/test/AbstractSearchTest.java (hunk 1 above)
Comment (adimiz1) [reply cloudinary-test-common/src/main/java/com/cloudinary/test/AbstractSearchTest.java]: Fixed

File: cloudinary-test-common/src/main/java/com/cloudinary/test/AbstractSearchTest.java [hunk 2]
```diff
@@ -44,6 +48,7 @@ public static void setUpClass() throws Exception {
     public static void tearDownClass() throws Exception {
         Cloudinary cloudinary = new Cloudinary();
         cloudinary.api().deleteResourcesByTag(SEARCH_TAG, null);
+        cloudinary.api().deleteFolder(SEARCH_FOLDER, null);
```
Comment (const-cloudinary) [inline cloudinary-test-common/src/main/java/com/cloudinary/test/AbstractSearchTest.java:51]: If the folder does not exist will it gracefully exit or crash?

**Review by const-cloudinary — CHANGES_REQUESTED** [review]

File: cloudinary-test-common/src/main/java/com/cloudinary/test/AbstractSearchTest.java (hunk 2 above)
Comment (adimiz1) [reply cloudinary-test-common/src/main/java/com/cloudinary/test/AbstractSearchTest.java]: Wrapped with try/catch

**Review by const-cloudinary — APPROVED** [review]

**Pull request merged**